runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -221,6 +226,86 @@ private ClusterStatus getClusterHealth(Layout layout, Set<String> peerResponsive
         return peerResponsiveNodes;
     }
 
+    /**
+     * Fetches the initial layout to fetch the cluster status.
+     * This does not consume the corfu runtime layout as that can get stuck in an indefinite retry
+     * loop.
+     *
+     * @param layoutServers Layout Servers to fetch the layout from.
+     * @return Latest layout from the servers or null if none of them responded with a layout.
+     */
+    private Layout getClusterStatusLayout(List<String> layoutServers) {
+        Layout layout = null;
+        Map<String, CompletableFuture<Layout>> layoutFuturesMap = new HashMap<>();
+        // Get layout futures for layout requests to all layout servers.
+        for (String server : layoutServers) {
+            try {
+                IClientRouter router = runtime.getRouter(server);
+                layoutFuturesMap.put(server,
+                        new LayoutClient(router, Layout.INVALID_EPOCH).getLayout());
+            } catch (Exception e) {
+                log.error("getClusterStatus: Exception encountered connecting to {}. ",
+                        server, e);
+                CompletableFuture<Layout> cf = new CompletableFuture<>();
+                cf.completeExceptionally(e);
+                layoutFuturesMap.put(server, cf);
+            }
+        }
+
+        // Wait on the Completable futures and retain the layout with the highest epoch.
+        for (Entry<String, CompletableFuture<Layout>> entry : layoutFuturesMap.entrySet()) {
+            try {
+                Layout nodeLayout = entry.getValue().get();
```

Comment:
What if this times out ?

Reply:
This is a best-effort API. So the result will be based on the layouts provided by the rest of the nodes. The timeout, however, does not declare the node as unresponsive yet.
We later heartbeat all the nodes with a certain number of retries which help in formulating the ClusterReport.

Reply:
What if it times out and the exception leaks?

Reply:
The exception is caught by the ExecutionException. All Exceptions(including RuntimeExceptions) are wrapped in the ExecutionException while blocking on a CompletableFutire callback.